orc/src/main/java/org/apache/iceberg/orc/OrcValueReaders.java

```diff
@@ -178,15 +171,29 @@ public T nonNullRead(ColumnVector vector, int row) {
     }
 
     private T readInternal(T struct, ColumnVector[] columnVectors, int row) {
-      for (int c = 0; c < readers.length; ++c) {
-        set(struct, c, reader(c).read(columnVectors[c], row));
+      for (int c = 0, vectorIndex = 0; c < readers.length; ++c) {
+        ColumnVector vector = isConstantField[c] ? null : columnVectors[vectorIndex++];
```

Comment:
Minor: we discourage using the return value of `++` expressions because it is error prone and makes code harder to read. Could you move the increment to a separate line?

Reply:
Done